test/libgit2.jl

```diff
@@ -1013,18 +1013,22 @@ mktempdir() do dir
             end
 
             # Find a hostname that maps to the loopback address
-            hostname = replace(readchomp(`hostname`), r"\..*$", "")
-            loopback = ip"127.0.0.1"
+            hostnames = ["localhost"]
+            try
+                # In minimal environments a hostname might not be available (issue #20758)
+                unshift!(hostnames, replace(gethostname(), r"\..*$", ""))
```

Comment:
The regex strips everything after the first dot ("a.b.c" -> "a"). Some environment (Macs) have "macbook.local" bound to the external address while "macbook" is bound to the loopback address.